orx-mesh-generators/src/commonMain/kotlin/Extrusion.kt

```diff
@@ -0,0 +1,242 @@
+package org.openrndr.extra.meshgenerators
+
+import org.openrndr.math.Matrix44
+import org.openrndr.math.Vector2
+import org.openrndr.math.Vector3
+import org.openrndr.math.Vector4
+import org.openrndr.math.transforms.normalMatrix
+import org.openrndr.shape.Path3D
+import org.openrndr.shape.Shape
+import org.openrndr.shape.ShapeContour
+import org.openrndr.shape.Triangle
+
+fun quadToTris(v00: Vector3, v01: Vector3, v10: Vector3, v11: Vector3, faceNormal: Vector3, writer: VertexWriter) {
+
+    writer(v11, faceNormal, Vector2.ZERO)
+    writer(v01, faceNormal, Vector2.ZERO)
+    writer(v00, faceNormal, Vector2.ZERO)
+
+    writer(v00, faceNormal, Vector2.ZERO)
+    writer(v10, faceNormal, Vector2.ZERO)
+    writer(v11, faceNormal, Vector2.ZERO)
+}
+
+fun contourSegment(linearContour: List<Vector3>, frame0: Matrix44, frame1: Matrix44, writer: VertexWriter) {
+    for (i in linearContour.indices) {
+        val v0 = linearContour[i]
+        val v1 = linearContour[(i + 1).mod(linearContour.size)]
+
+        val v00 = (frame0 * v0.xyz1).xyz
+        val v01 = (frame0 * v1.xyz1).xyz
+
+        val v10 = (frame1 * v0.xyz1).xyz
+        val v11 = (frame1 * v1.xyz1).xyz
+        val faceNormal = ((v10 - v00).normalized cross (v01 - v00).normalized).normalized
```

Comment:
Do you have in mind also having `vertexNormal` as an alternative? (for smooth instead of low-poly look)

Reply:
Eventually yes. I am doubthing between computing the vertex normals from weighted(?) face normals and evaluating the vertex normals locally. The latter may not be possible if variable scaling or offsetting is introduced.